### PR TITLE
File info fix

### PR DESCRIPTION
**Pull request by hackrush01**

### CODE REVIEW

File: ui/js/selectors/file_info.js [hunk 1]
```diff
@@ -129,14 +140,14 @@ export const selectFileInfosByUri = createSelector(
...
 
 export const selectDownloadingFileInfos = createSelector(
-  selectDownloadingByOutpoint,
-  selectFileInfosByOutpoint,
-  (downloadingByOutpoint, fileInfosByOutpoint) => {
-    const outpoints = Object.keys(downloadingByOutpoint);
+  selectDownloadingBySdHash,
+  selectFileInfosBySdHash,
+  (downloadingBySdHash, fileInfosBySdHash) => {
+    const sd_hashes = Object.keys(downloadingBySdHash);
```
Comment (kauffj) [inline ui/js/selectors/file_info.js]: `sdHashes`

File: ui/js/selectors/file_info.js [hunk 2]
```diff
@@ -110,16 +121,16 @@ export const selectFileInfosPublished = createSelector(
...
+  (claimsByUri, bySdHash) => {
     const fileInfos = {};
     const uris = Object.keys(claimsByUri);
 
     uris.forEach(uri => {
       const claim = claimsByUri[uri];
       if (claim) {
-        const outpoint = `${claim.txid}:${claim.nout}`;
-        const fileInfo = byOutpoint[outpoint];
+        const sd_hash = claim.value.stream.source.source;
```
Comment (kauffj) [inline ui/js/selectors/file_info.js]: `.source.source`? Guessing this is correct, but please tell me who to complain to :P

File: ui/js/selectors/claims.js [hunk 3]
```diff
@@ -185,6 +185,22 @@ export const selectMyClaimsOutpoints = createSelector(
   }
 );
 
+export const selectMyClaimSdHashesByOutpoint = createSelector(
```
Comment (kauffj) [inline ui/js/selectors/claims.js]: Can you help me understand why this is necessary? When do I have an outpoint that I need to translated into an sd_hash? Can we just stored sd_hashes in that place instead?

File: ui/js/selectors/claims.js [hunk 4]
```diff
@@ -185,6 +185,22 @@ export const selectMyClaimsOutpoints = createSelector(
...
+  selectMyClaims,
+  myClaims => {
+    const sdHashByOutpoint = {};
+
+    console.log(myClaims);
```
Comment (kauffj) [inline ui/js/selectors/claims.js]: *ahem*

File: ui/js/actions/content.js [hunk 5]
```diff
@@ -187,55 +188,57 @@ export function doUpdateLoadStatus(uri, outpoint) {
...
-            dispatch(doUpdateLoadStatus(uri, outpoint));
+            dispatch(doUpdateLoadStatus(uri, outpoint, sd_hash));
           }, DOWNLOAD_POLL_INTERVAL);
         }
       });
   };
 }
 
-export function doStartDownload(uri, outpoint) {
+export function doStartDownload(uri, outpoint, sd_hash) {
```
Comment (kauffj) [inline ui/js/actions/content.js]: Can this be re-written to only use `sd_hash` rather than both `outpoint` and `sd_hash`?

Comment (hackrush01) [reply ui/js/actions/content.js]: TL;DR I'm not sure.
It went a bit too deep inside `lbry.js` and it seemed to be intertwined with other stuff(such as pending publishes, dummy claims). Overwhelmed by the complications involved I couldn't muster up enough courage to mess with it.

File: ui/js/selectors/claims.js (hunk 3 above)
Comment (hackrush01) [reply ui/js/selectors/claims.js]: I think it is necessary due to what happens when I have a pending publish(related to https://github.com/lbryio/lbry-app/pull/735#discussion_r151756127 probably).

File: ui/js/selectors/claims.js (hunk 4 above)
Comment (hackrush01) [reply ui/js/selectors/claims.js]: Oops!

File: ui/js/selectors/file_info.js (hunk 2 above)
Comment (hackrush01) [reply ui/js/selectors/file_info.js]: The schema/lbryum/daemon guys.

File: ui/js/selectors/file_info.js (hunk 1 above)
Comment (hackrush01) [reply ui/js/selectors/file_info.js]: Again, Oops!

**Comment by tzarebczan** [comment]
Some scenarios to test alongside these changes:
https://github.com/lbryio/lbry/issues/951
https://github.com/lbryio/lbry/issues/944
https://github.com/lbryio/lbry-app/issues/600

**Comment by kauffj** [comment]
I'm not planning to merge this without reducing the amount of values carried around from 2 to 1. If `sd_hash` is the new key for file data instead of outpoint, let's eliminate the usage of outpoint basically everywhere.

**Comment by hackrush01** [comment]
I'll look into it and rebase it on master soon.

**Comment by kauffj** [comment]
@hackrush01 I've actually made a bunch of progress on this separately in the `flow_and_file_list` branch (I used it as my test for writing code typed with Flow). I'll finish this.

**Comment by hackrush01** [comment]
Ok

**Comment by lyoshenka** [comment]
blocked by lbryio/lbry#1025

**Comment by kauffj** [comment]
This is no longer necessary, @liamcardenas has handled this.

**Pull request closed without merging**